scionlab/models/core.py

```diff
@@ -650,13 +651,20 @@ def create(self,
         :param str bind_ip: optional, the bind IP for this interface to override host.bind_ip.
         :param int bind_port: optional, a free port is selected if bind IP set and not specified
         :param int interface_id: optional, the interface id for this interface.
+        :param VPNClient vpn_client: the `vpn_client` associated to this Interface (only used to
+                                     avoid hitting the DB to retrieve it)
         """
         host = border_router.host
         as_ = host.AS
         ifid = interface_id or as_.find_interface_id()
 
         effective_public_ip = public_ip or host.public_ip
-        effective_bind_ip = bind_ip if public_ip else host.bind_ip
+        if vpn_client is not None:
+            # If using VPN the `public_ip` is the one of the VPNClient and `bind_ip` is None
+            public_ip = vpn_client.ip
+            effective_bind_ip = None
```

Comment:
Removing this piece of information means losing track of which links are over VPN.
If we really want to do this, it might still be possible to infer wheter a `Link` (and thus an `Interface`) is over VPN by looking at the IP of the interface, and checking it against the `VPNClient`'s IP, and in case they are the same then we can assume the connection is over vpn.
This shouldn't hit the DB more than once actually.